backend/fms_core/services/event_file.py

```diff
@@ -0,0 +1,233 @@
+from typing import Iterable, List, TextIO, Union
+from dataclasses import asdict, dataclass
+from io import StringIO
+import json
+from fms_core.models import (
+    Biosample, 
+    DerivedSample, 
+    DerivedBySample,
+    ExperimentRun, 
+    Index,
+    Individual,
+    Instrument, 
+    Library,
+    ProcessMeasurement,
+    Project,
+    PropertyValue,
+    Protocol,
+    Sample, 
+    SampleLineage,
+)
+
+FMS_Id = Union[int, None]
+
+@dataclass
+class EventSample:
+    fms_sample_name: Union[str, None]
+
+    fms_sample_id: FMS_Id = None
+    fms_derived_sample_id: FMS_Id = None
+    fms_biosample_id: FMS_Id = None
+
+    # Flowcell lane containing the sample
+    fms_container_coordinates: Union[str, None] = None
+
+    fms_project_id: FMS_Id = None
+    fms_project_name: Union[str, None] = None
+    hercules_project_id: Union[str, None] = None
+    hercules_project_name: Union[str, None] = None
+
+    fms_pool_volume_ratio: Union[float, None] = None
+
+    # individual fields
+    fms_expected_sex: Union[str, None] = None
+    ncbi_taxon_id: Union[int, None] = None
+    fms_taxon_name: Union[str, None] = None
+    
+    # library fields
+    fms_platform_name: Union[str, None] = None
+    fms_library_type: Union[str, None] = None
+    fms_library_size: Union[float, None] = None
+    fms_index_id: FMS_Id = None
+    fms_index_name: Union[str, None] = None
+    fms_index_sequence_3_prime: Union[List[str], None] = None
+    fms_index_sequence_5_prime: Union[List[str], None] = None
+    fms_library_kit: Union[str, None] = None
+    
+
+@dataclass
+class EventManifest:
+    # Manifest version (1.0.0 to start) 
+    version: str
+
+    # Experiment run name and id
+    run_name: str
+    fms_run_id: int
+    fms_run_start_date: Union[str, None]
+
+    # Flowcell / container for experiment
+    fms_container_id: int
+    fms_container_barcode: str
+
+    instrument_id: str
+    fms_instrument_type: str
+
+    samples: List[EventSample]
+
+
+EVENT_FILE_VERSION = "1.0.0"
```

Comment:
please get rid of any event file references. Let's give it a more useful name. Run info file. fms run export file.